File: src/com/darin/weex/ui/preview/WeexEditorProvider.java
package com.darin.weex.ui.preview;

import com.darin.weex.language.WeexFileType;
import com.darin.weex.language.WeexLanguage;
import com.intellij.openapi.fileEditor.FileEditor;
import com.intellij.openapi.fileEditor.FileEditorPolicy;
import com.intellij.openapi.fileEditor.FileEditorProvider;
import com.intellij.openapi.fileEditor.FileEditorState;
import com.intellij.openapi.project.PossiblyDumbAware;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.jdom.Element;
import org.jetbrains.annotations.NotNull;

/**
 * Created by darin on 5/18/16.
 */
public class WeexEditorProvider implements FileEditorProvider, PossiblyDumbAware {
    public static final String EDITOR_TYPE_ID = WeexLanguage.NAME + "FxPreviewEditor";

    @Override
    public boolean accept(@NotNull Project project, @NotNull VirtualFile virtualFile) {
        return WeexFileType.isWeexLanguage(virtualFile);
    }

    @NotNull
    @Override
    public FileEditor createEditor(@NotNull Project project, @NotNull VirtualFile virtualFile) {
        return new WeexFxPreviewEditor(project, virtualFile);
    }

    @Override
    public void disposeEditor(@NotNull FileEditor fileEditor) {

    }

    @NotNull
    @Override
    public FileEditorState readState(@NotNull Element element, @NotNull Project project, @NotNull VirtualFile virtualFile) {
        return FileEditorState.INSTANCE;
    }

    @Override
    public void writeState(@NotNull FileEditorState fileEditorState, @NotNull Project project, @NotNull Element element) {

    }

    @NotNull
    @Override
    public String getEditorTypeId() {
        return EDITOR_TYPE_ID;
    }

    @NotNull
    @Override
    public FileEditorPolicy getPolicy() {
        return FileEditorPolicy.HIDE_DEFAULT_EDITOR;
    }

    @Override
    public boolean isDumbAware() {
        return true;
    }
}

File: src/com/darin/weex/ui/preview/WeexEditorProvider.kt
package com.darin.weex.ui.preview

import com.darin.weex.language.WeexFileType
import com.darin.weex.language.WeexLanguage
import com.darin.weex.utils.WeexConstants
import com.intellij.openapi.fileEditor.FileEditor
import com.intellij.openapi.fileEditor.FileEditorPolicy
import com.intellij.openapi.fileEditor.FileEditorProvider
import com.intellij.openapi.fileEditor.FileEditorState
import com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorProvider
import com.intellij.openapi.project.PossiblyDumbAware
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile
import org.jdom.Element

/**
 * Created by darin on 5/18/16.
 */
class WeexEditorProvider : FileEditorProvider, PossiblyDumbAware {

    override fun accept(project: Project, virtualFile: VirtualFile): Boolean {
        return WeexFileType.isWeexLanguage(virtualFile)
    }

    override fun createEditor(project: Project, virtualFile: VirtualFile): FileEditor = WeexFxPreviewEditor(project, virtualFile)

    override fun disposeEditor(fileEditor: FileEditor) {

    }

    override fun readState(element: Element, project: Project, virtualFile: VirtualFile): FileEditorState {
        return FileEditorState.INSTANCE
    }

    override fun writeState(fileEditorState: FileEditorState, project: Project, element: Element) {

    }

    override fun getEditorTypeId(): String {
        return EDITOR_TYPE_ID
    }

    override fun getPolicy(): FileEditorPolicy {
        return FileEditorPolicy.HIDE_DEFAULT_EDITOR
    }

    override fun isDumbAware(): Boolean {
        return true
    }

    companion object {
        val EDITOR_TYPE_ID = WeexLanguage.NAME + "FxPreviewEditor"
    }
}
